index.js

```diff
@@ -55,6 +89,13 @@ module.exports = robot => {
           }),
         );
       }
+
+      // not proud of this, but only way to ensure failing assertions
+      // throw, while also allowing this to not exit the process in
+      // production
+      if (global.TESTING) {
+        throw err;
```

Comment:
Where do we assert on the failure? Just not sure this is needed if we can inspect the behavior instead (like doing an expect() on the content that posts to github, or mocking the createComment() api.

Reply:
It's [the test assertion here](https://github.com/uber-workflow/probot-app-merge-pr/pull/77/files#diff-518498ef0535a2e4c194b95d1cff9c3fR48). I totally agree that it's ugly, but since the assertion is in the network request, it gets caught by this try/catch

Reply:
I'm not sure how straightforward it would be to mock the octokit methods since they're provided by probot inside the app instance (see [this line](https://github.com/uber-workflow/probot-app-merge-pr/pull/77/files#diff-168726dbe96b3ce427e7fedce31bb0bcL19))

Reply:
But this is for the failure case, correct? That assertion seems to not be for a failing case?

Reply:
Yes, but if the assertion fails, then it gets eaten up and jest reports it as a success (I spent like an hour trying to figure out why jest was reporting a false positive for this, lol)

Reply:
Ooh, I suppose because the `expect()` is inside of a callback. That makes sense now. I wonder if we should use async/await for this kind of stuff, makes it easier to reason about.